client/src/components/App/Content/LandingPage/InvestigationTable/InvestigatorAllocation/InvestigatorsTable/InvestigatorsTable.tsx

```diff
@@ -19,45 +22,89 @@ const InvestigatorsTable: React.FC<Props> = ({ investigators, selectedRow, setSe
 
     const [investigatorInput, setInvestigatorInput] = useState<string>('');
     const [filteredInvestigators, setFilteredInvestigators] = useState<User[]>(investigators);
+    const [sortedInvestigators, setStortedInvestigators] = useState<User[]>(filteredInvestigators);
+    const [order, setOrder] = useState<Order>(SortOrder.asc);
+    const [orderBy, setOrderBy] = useState<string>(defaultOrderBy);
+    const [orderByValue, setOrderByValue] = useState<string>(defaultOrderBy);
 
     useEffect(() => {
         setFilteredInvestigators(investigators)
+        setStortedInvestigators(investigators)
     }, [investigators]);
-    
+
     useEffect(() => {
-        if(investigatorInput !== '') {
-            const filteredArray = investigators.filter(investigator => investigator.userName.includes(investigatorInput) || 
-                                                                    (investigator.authorityName && investigator?.authorityName.includes(investigatorInput)) ||
-                                                                    (investigator.sourceOrganization && investigator?.sourceOrganization.includes(investigatorInput)));
-            setFilteredInvestigators(filteredArray)   
+        if (investigatorInput !== '') {
+            const filteredArray = investigators.filter(investigator => investigator.userName.includes(investigatorInput) ||
+                (investigator.authorityName && investigator?.authorityName.includes(investigatorInput)) ||
+                (investigator.sourceOrganization && investigator?.sourceOrganization.includes(investigatorInput)));
+            setFilteredInvestigators(filteredArray)
+            sortInvestigators(filteredInvestigators)
         } else {
             setFilteredInvestigators(investigators)
+            setStortedInvestigators(investigators)
+            sortInvestigators(investigators)
         }
     }, [investigatorInput]);
 
+    useEffect(() => {
+        sortInvestigators(filteredInvestigators)
+        }, [orderByValue]);
+
+    const sortInvestigators = (investigatorsToOrder: User[]) => {
+        switch(orderBy) {
+            case 'username': {
+                const orderd = _.orderBy(investigatorsToOrder, [investigator => investigator['userName']], [order])
+                setStortedInvestigators(orderd)
+                break;
+            }
+            case 'sourceorganization': {
+                const orderd = _.orderBy(investigatorsToOrder, [investigator => investigator['sourceOrganization']], [order])
+                setStortedInvestigators(orderd)
+                break;
+            }
+            case 'deskname': {
+                const orderd = _.orderBy(investigatorsToOrder, [investigator => investigator['deskname']], [order])
+                setStortedInvestigators(orderd)
+                break;
+            }
+            default: {
+                setStortedInvestigators(investigatorsToOrder)
+                break;
+            }
+        }
+    }
```

Comment:
cant you use a function like : 
const foo = (orderBy) => {
    const orderd = _.orderBy(investigatorsToOrder, [investigator => investigator[orderedBy]], [order])
    setStortedInvestigators(orderd)
}
?